### PR TITLE
add search history,expandcollapse to search operation

### PR DESCRIPTION
**Pull request by arun3688**

### CODE REVIEW

File: OMEdit/OMEditGUI/Search/SearchWidget.cpp [hunk 1]
```diff
@@ -166,9 +185,26 @@ void SearchWidget::searchInFiles()
...
+  connect(mpSearch, SIGNAL(setProgressBarValue(int,int)), mpSearchWidgetResult, SLOT(updateProgressBarValue(int,int)));
+  connect(mpSearch, SIGNAL(setFoundFilesLabel(int)), mpSearchWidgetResult, SLOT(updateFoundFilesLabel(int)));
+  connect(mpSearch, SIGNAL(setProgressBarCancelValue(int,int)), mpSearchWidgetResult, SLOT(updateProgressBarCancelValue(int,int)));
+  connect(mpSearchWidgetResult, SIGNAL(setCancelSearchResult()), mpSearch, SLOT(updateCancelSearch()));
+  /*emit the signals to stop any ongoing search operation when mainwindow is closed*/
+  connect(this, SIGNAL(setCancelSearch()), mpSearch, SLOT(updateCancelSearch()));
+
+  mpSearchStackedWidget->addWidget(mpSearchWidgetResult);
+  mpSearchStackedWidget->setCurrentWidget(mpSearchWidgetResult);
+  QString searchHistoryItem = QString("%1%2%3 %4").arg("Project-").arg(mpSearchScopeComboBox->currentText()).arg(":").arg(mpSearchStringComboBox->currentText());
```
Comment (adeas31) [inline OMEdit/OMEditGUI/Search/SearchWidget.cpp]: Why not,
`QString("%1-%2: %3").arg(tr("Project")).arg(mpSearchScopeComboBox->currentText()).arg(mpSearchStringComboBox->currentText());`

File: OMEdit/OMEditGUI/Search/SearchWidget.h [hunk 2]
```diff
@@ -42,41 +42,68 @@
...
   SearchWidget(QWidget *pParent = 0);
   ~SearchWidget();
-  QProgressBar *getProgressBar() {return mpProgressBar;}
   QComboBox *getSearchScopeComboBox() {return mpSearchScopeComboBox;}
   QComboBox *getSearchStringComboBox() {return mpSearchStringComboBox;}
   QComboBox *getSearchFilePatternComboBox() {return mpSearchFilePatternComboBox;}
+  QStackedWidget *getSearchStackedWidget() {return mpSearchStackedWidget;}
+  QComboBox * getSearchHistoryCombobox() {return mpSearchHistoryComboBox;}
   void updateComboBoxSearchStrings(QComboBox *pComboBox);
+  QList<SearchWidgetResult*> pSearchWidgetResultobjects;
```
Comment (adeas31) [inline OMEdit/OMEditGUI/Search/SearchWidget.h]: This is a list not a pointer. So it should be `QList<SearchWidgetResult*> mSearchWidgetResultobjects;`

File: OMEdit/OMEditGUI/Search/SearchWidget.h [hunk 3]
```diff
@@ -42,41 +42,68 @@
...
+  QComboBox *mpSearchHistoryComboBox;
+  QStackedWidget *mpSearchStackedWidget;
+  QAction * mpClearAction;
+  QAction * mpExpandAction;
+  QAction * mpCollapseAction;
+  SearchWidgetResult * mpSearchWidgetResult;
+  Search *mpSearch;
+};
+
+class SearchWidgetResult : public QWidget
```
Comment (adeas31) [inline OMEdit/OMEditGUI/Search/SearchWidget.h]: A proper name would be **SearchResultWidget**.

**Review by adeas31 — CHANGES_REQUESTED** [review]
Just few minor fixes required.

**Comment by arun3688** [comment]
@adeas31 , fixed the changes

**Comment by adeas31** [comment]
* I still see the text **Searching 1093 of 1093 files. Please wait for a while** after the search is finished.
* Unable to cancel the search.
* Pressing `ctrl+h` doesn't show the Search Browser when I tab it in the dock widget.

**Comment by arun3688** [comment]
Unable to cancel the search.?, Are you not able to cancel the search, because i tested it and 
I am able to cancel the search,

**Comment by arun3688** [comment]
updated the changes

**Review by adeas31 — APPROVED** [review]

**Pull request merged**